collector/generic_collector.go

```diff
@@ -160,6 +172,12 @@ func (collector *GenericCollector) Collect(metrics map[string][]v1.MetricVal) (t
 		} else {
 			errorSlice = append(errorSlice, fmt.Errorf("No match found for regexp: %v for metric '%v' in config", metricConfig.Regex, metricConfig.Name))
 		}
+		if len(newMetrics) > collector.info.metricCountLimit {
```

Comment:
Probably no, but I placed this check to be consistent with the prometheus code. Can be removed if you wish.